### DISCUSSION TITLE
Any way to disable `corfu-sort-function` temporarily?

### DISCUSSION THREAD

**Discussion by xofyarg** [minad/corfu#507] [Q&A]
Hi, I'd like to write a capf, and use it with corfu. For this completion source, I don't want the the candidates to be sorted, as it's something about recent command history. But seems like I cannot do something like `(let ((corfu-sort-function nil)) (completion-at-point))`, because corfu does something in the `post-command-hook`. Another approach is to emit `display-sort-function` from the source, but it's a bit cumbersome to implement, and also a bit slower, because the collection table function get called several times.

I wonder if I missed anything, and there are some options for it. Thanks.

**Comment by jdtsmith** [comment]
You could configure your CAPF itself to emit `display-sort-function` among its `metadata` (setting to `identity`).  If you have no control over the CAPF, you could instead use `cape-wrap-properties` to wrap it and add the relevant properties, including a builtin `:sort nil` keyword you can use to disable sorting.
